### ISSUE TITLE
Include directive with operation argument errors out in Fed 2.1

### ISSUE THREAD

**Issue by smyrick** [apollographql/federation#2124]
Example code: https://stackblitz.com/edit/apollo-basic-fed2-demo-ixjqpb?file=README.md

### Schema
```graphql
# service A
type Query {
  a: A
}
type A @key(fields: "id") {
  id: ID!
  foo: String
}
```

```graphql
# service B
type A @key(fields: "id") {
  id: ID!
  bar: String
}
```

### Query

Run the following query

```graphql
query Test($if: Boolean!) {
  a {
    id
    foo
    ...BSelect @include(if: $if)
  }
}

fragment BSelect on A {
  bar
}
```

on the following input it runs fine

```json
{
  "if": false
}
```

on setting the value to `true` it errors out

```json
{
  "if": true
}
```

```json
{
  "message": "400: Bad Request",
  "extensions": {
    "response": {
      "url": "http://localhost:4002/graphql",
      "status": 400,
      "statusText": "Bad Request",
      "body": {
        "errors": [
          {
            "message": "Variable \"$if\" is never used in operation \"Test__b__1\".",
            "extensions": {
              "code": "GRAPHQL_VALIDATION_FAILED",
              "exception": {
                "stacktrace": [
                  "GraphQLError: Variable \"$if\" is never used in operation \"Test__b__1\".",
                  "    at Object.leave (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/graphql/validation/rules/NoUnusedVariablesRule.js:39:15)",
                  "    at Object.leave (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/graphql/language/visitor.js:324:32)",
                  "    at Object.leave (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/graphql/utilities/TypeInfo.js:411:21)",
                  "    at visit (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/graphql/language/visitor.js:197:21)",
                  "    at validate (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/graphql/validation/validate.js:91:24)",
                  "    at validate (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/apollo-server-core/dist/requestPipeline.js:186:39)",
                  "    at processGraphQLRequest (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/apollo-server-core/dist/requestPipeline.js:98:34)",
                  "    at async processHTTPRequest (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/apollo-server-core/dist/runHttpQuery.js:220:30)"
                ]
              }
            }
          }
        ]
      }
    },
    "code": "INTERNAL_SERVER_ERROR",
    "exception": {
      "stacktrace": [
        "GraphQLError: 400: Bad Request",
        "    at RemoteGraphQLDataSource.errorFromResponse (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/datasources/RemoteGraphQLDataSource.js:169:16)",
        "    at async RemoteGraphQLDataSource.sendRequest (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/datasources/RemoteGraphQLDataSource.js:103:23)",
        "    at async RemoteGraphQLDataSource.process (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/datasources/RemoteGraphQLDataSource.js:79:26)",
        "    at async sendOperation (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/executeQueryPlan.js:232:26)",
        "    at async eval (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/executeQueryPlan.js:194:49)",
        "    at async executeNode (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/executeQueryPlan.js:130:17)",
        "    at async executeNode (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/executeQueryPlan.js:121:27)",
        "    at async executeNode (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/executeQueryPlan.js:104:40)",
        "    at async eval (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/executeQueryPlan.js:30:35)",
        "    at async eval (/home/projects/apollo-basic-fed2-demo-ixjqpb/node_modules/@apollo/gateway/dist/index.js:105:38)"
      ]
    }
  }
}
```

**Comment by pcmanus** [comment]
The underlying problem is that due to an oversight from #1911, directives applied to a fragment spread were lost in subgraph queries (this shows up as the validation error seen above because, due to how the code is organised, the test for whether a variable should be included in the subgraph query or not was done before the directive was mistakenly lost, and so the subgraph query did end up with a non-used variable; but of course, the issue is not that the variable is included, but rather than the directive using it is not). Fixed pushed at #2126.

**Comment by abernix** [comment]
I'll try to track this landing in the Router here: https://github.com/apollographql/router/issues/1698
